src/rust/bitbox02-rust/src/hww/api/ethereum/sign.rs

```diff
@@ -1175,4 +1251,131 @@ mod tests {
+            nonce: b"\x1f\xdc".to_vec(),
+            gas_price: b"\x01\x65\xa0\xbc\x00".to_vec(),
+            gas_limit: b"\x52\x08".to_vec(),
+            recipient:
+                b"\x04\xf2\x64\xcf\x34\x44\x03\x13\xb4\xa0\x19\x2a\x35\x28\x14\xfb\xe9\x27\xb8\x85"
+                    .to_vec(),
+            value: b"\x07\x5c\xf1\x25\x9e\x9c\x40\x00".to_vec(),
+            data: b"".to_vec(),
+            host_nonce_commitment: None,
+            chain_id: 42161,
+            address_case: pb::EthAddressCase::Mixed as _,
+        })))
+        .unwrap();
+        assert_eq!(unsafe { CONFIRM_COUNTER }, 1);
+    }
+
+    /// Test that EIP-1559 transactions also get the chain confirmation screen
+    #[test]
+    pub fn test_chain_confirmation_for_eip1559() {
+        const KEYPATH: &[u32] = &[44 + HARDENED, 60 + HARDENED, 0 + HARDENED, 0, 0];
+        static mut CONFIRM_COUNTER: u32 = 0;
+
+        // Test with Polygon network (chain_id 137)
+        mock(Data {
+            ui_confirm_create: Some(Box::new(|params| {
+                unsafe {
+                    if CONFIRM_COUNTER == 0 {
+                        assert_eq!(params.body, "Sign transaction on\n\nPolygon");
+                        CONFIRM_COUNTER += 1;
+                    }
+                }
+                true
+            })),
+            ui_transaction_address_create: Some(Box::new(|_, _| true)),
+            ui_transaction_fee_create: Some(Box::new(|_, _, _| true)),
+            ..Default::default()
+        });
+        mock_unlocked();
+
+        block_on(process(&Transaction::Eip1559(&pb::EthSignEip1559Request {
+            keypath: KEYPATH.to_vec(),
+            nonce: b"\x1f\xdc".to_vec(),
+            max_priority_fee_per_gas: b"\x3b\x9a\xca\x00".to_vec(),
+            max_fee_per_gas: b"\x01\x65\xa0\xbc\x00".to_vec(),
+            gas_limit: b"\x52\x08".to_vec(),
+            recipient:
+                b"\x04\xf2\x64\xcf\x34\x44\x03\x13\xb4\xa0\x19\x2a\x35\x28\x14\xfb\xe9\x27\xb8\x85"
+                    .to_vec(),
+            value: b"\x07\x5c\xf1\x25\x9e\x9c\x40\x00".to_vec(),
+            data: b"".to_vec(),
+            host_nonce_commitment: None,
+            chain_id: 137,
+            address_case: pb::EthAddressCase::Mixed as _,
+        })))
+        .unwrap();
+        assert_eq!(unsafe { CONFIRM_COUNTER }, 1);
+    }
+
+    /// Test that unknown networks do NOT get the chain confirmation screen
+    #[test]
+    pub fn test_no_chain_confirmation_for_unknown_networks() {
+        const KEYPATH: &[u32] = &[44 + HARDENED, 60 + HARDENED, 0 + HARDENED, 0, 0];
+        static mut CONFIRM_COUNTER: u32 = 0;
+
+        // Test with an unknown network (chain_id 999999)
+        mock(Data {
+            ui_confirm_create: Some(Box::new(|params| {
+                unsafe {
+                    // Only the warning confirmations should appear, not chain confirmation
+                    if params.body.starts_with("Sign transaction on") {
```

Comment:
this test is a bit brittle, as if the wording changes it wouldn't catch a regression. would be safer to check all expected screens.

Reply:
actually, looking at existing tests, this one is not needed since "test_process_unknown_network" will already catch this, removed